editor/visual_script_editor.cpp

```diff
@@ -3663,7 +3650,7 @@ void VisualScriptEditor::_selected_connect_node(const String &p_text,
 
 	Ref<VisualScriptNode> vnode;
 	Ref<VisualScriptNode> vnode_old;
-	bool port_node_exists = script->has_node(port_action_node);
+	bool port_node_exists = true;
```

Comment:
Will a port always exist?

Reply:
There is a two types of nodes in visual scripting.

1. sequenced nodes. Usually if statements etc.
2. unsequenced nodes. Usually variable get sets

Reply:
I understand the situation you described, but during my testing, I found that if I don't make changes here, the following situation will occur: Normally, when I pull a connection curve from a node that has already been created, a search box will pop up, and when I enter a letter in the search box, the corresponding tree search results will be displayed. However, when it is "bool port_node_exists = script->has_node(port_action_node);", it cannot display the tree search results at this time, and it must first press the "search classes" option to display the tree search results. Even if I close the "search classes" option, it can still display the tree search results. But when I redo the above operation, I must open the "search classes" option again, and then close it before it can display the tree search results again. This problem is indeed somewhat puzzling. So I tried to change it to "bool port_node_exists = true;", and found that this change restored normal operation, without the need to open and close the "search classes" option once, and it can display the tree search results normally. Maybe my modification is somewhat unreasonable, and it is only a temporary solution to this problem. Currently, I have not found any other way to handle the above problem.
Based on the situation I described above, what effective suggestions do you have?

Reply:
We can discuss how to handle this issue more reasonably, without rushing to merge it into the main branch until the issue is resolved properly.

Reply:
That is odd. Will need to think on it and do some experimentation.

Reply:
I think we can consider testing the return value of 'script-> has_node (port_action_node)' to find out the relationship between this return value and 'Search classes'. Perhaps the initial value of its return value may be 'false'.